sdks/python/apache_beam/transforms/core.py

```diff
@@ -2364,6 +2364,22 @@ def from_runner_api_parameter(
   def runner_api_requires_keyed_input(self):
     return True
 
+  def get_windowing(self, inputs):
+    # Circular dep, because trigger pulls in CombineFn
+    from apache_beam.transforms.trigger import AfterProcessingTime
```

Comment:
This doesn't seem to handle more complex triggers, such as `AfterWatermark(early=AfterProcessingTime())`. How is that handled in Java? 

Alternatively, we could add continuation_windowing which would call continuation_trigger and we wouldn't have to special case this knowledge in GBK. (If we're moving away from the idea of continuation triggers in general, maybe that's not preferable.)

Reply:
Wow, I really must have written this in a hurry. Of course you are right. In Java every trigger has a continuation trigger and it is evaluated by a visitor: https://github.com/apache/beam/blob/4ff08fea21fb79200cd47d1e887b24568a644f7c/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Trigger.java#L103

Gonna convert this back to a draft and do it right.